core/preconditioner/jacobi.cpp

```diff
@@ -201,12 +204,25 @@ void Jacobi<ValueType, IndexType>::detect_blocks(
 
 
 template <typename ValueType, typename IndexType>
-void Jacobi<ValueType, IndexType>::generate(const LinOp *system_matrix)
+void Jacobi<ValueType, IndexType>::generate(const LinOp *system_matrix,
+                                            bool skip_sorting)
 {
     GKO_ASSERT_IS_SQUARE_MATRIX(system_matrix);
+    using csr_type = matrix::Csr<ValueType, IndexType>;
     const auto exec = this->get_executor();
-    const auto csr_mtx = copy_and_convert_to<matrix::Csr<ValueType, IndexType>>(
-        exec, system_matrix);
+    decltype(copy_and_convert_to<csr_type>(exec, system_matrix)) csr_mtx{};
```

Comment:
nit: Wouldn't it be shorter to be explicit here, i.e., `std::unique_ptr<csr_type>`?

Reply:
No because it is not a default `unique_ptr`. `unique_ptr<csr_type, std::function<void(R*)>>` would be accurate, but in my opinion more complicated (and potentially loses the information why we need this type). That is the reason why I use `decltype`.

Reply:
I see. In case we had a separate function taking care of the copy and/or sorting, we could replace this by `auto`, right?

Reply:
Yes, that is what I am working on right now (and I will definitively replace this with `auto` after I am done).

Reply:
Perfect, thanks :)